src/tqdm_publisher/_demo/_server.py

```diff
@@ -0,0 +1,89 @@
+
+    Defaults are chosen for a deterministic and regular update period of one second for a total time of one minute.
+    """
+    all_task_durations_in_seconds = [1.0 for _ in range(60)]  # One minute at one second per update
+    progress_bar = TQDMPublisher(iterable=all_task_durations_in_seconds)
+
+    def run_function_on_progress_update(format_dict: dict) -> None:
+        """
+        This is the injected callback that will be run on each update of the TQDM object.
+
+        Its first and only positional argument must be the `format_dict` of the TQDM instance. Additional customization
+        on outside parameters must be achieved by defining those fields at an outer scope and defining this
+        server-specific callback inside the local scope.
+
+        In this demo, we will execute the `client_callback` whose protocol is known only to the WebSocketHandler.
+        It has
+        """
+        client_callback(client_id=client_id, progress_bar_id=progress_bar_id, format_dict=format_dict)
+
+    progress_bar.subscribe(callback=run_function_on_progress_update)
+
+    for task_duration in progress_bar:
+        time.sleep(task_duration)
+
+
+class WebSocketHandler:
+    """Describe this class."""
+
+    def __init__(self) -> None:
+        """Initialize the mapping of client IDs to ."""
+        self.clients: Dict[str, Any] = dict()
+
+    def function_to_run_on_progress_update(self, *, client_id: str, progress_bar_id: str, format_dict: dict) -> None:
+        """This is..."""
+        asyncio.run(self.send(client_id=client_id, data=dict(progress_bar_id=progress_bar_id, format_dict=format_dict)))
+
+    async def send(self, client_id: str, data: dict) -> None:
+        """Send an arbitrary JSON object `data` to the client identifier by `client_id`."""
+        await self.clients[client_id].send(json.dumps(obj=data))
+
+    async def handler(self, websocket) -> None:
+        """Describe what the handler does."""
+        client_id = str(uuid4())
+        self.clients[client_id] = websocket  # Register client connection
+
+        try:
+            async for message in websocket:
+                message_from_client = json.loads(message)
+
+                if message_from_client["command"] == "start":
+                    thread = threading.Thread(
+                        target=start_progress_bar,
+                        kwargs=dict(
+                            client_id=client_id,
+                            progress_bar_id=message_from_client["progress_bar_id"],
+                            client_callback=self.function_to_run_on_progress_update,
+                        ),
+                    )
+                    thread.start()
+        finally:  # This is called when the connection is closed
+            if client_id in self.clients:
+                del self.clients[client_id]
+
+
+async def spawn_server() -> None:
+    """Spawn the server asynchronously."""
```

Comment:
Likewise, does `asyncio.Future` return anything after it awaits?

Reply:
Technically it can—though in this case it won't

The result of an asyncio.Future object is set by calling its `.set_result(result)` method. This method is typically called by the code that executes the asynchronous operation to indicate that the operation is done and to provide its result. 

Here, though, we aren't providing the future object to be called anywhere.